### PR TITLE
Add Chrome nightly (Chromium trunk) to wpt

### PR DESCRIPTION
**Pull request by Hexcles**
Chrome does not really have an official "nightly" channel. We are using it to refer to the Chromium trunk (tip of the tree) build (available via https://download-chromium.appspot.com/).

This PR enables us to
* `wpt install [--download-only] [--channel nightly] chrome browser` and `wpt install [--channel nightly] chrome webdriver`
* `wpt run --install-browser --channel nightly chrome` (experimental web platform features will be enabled, same as dev or canary)

Note that this changes the default / latest channel for Chrome to nightly when no channel is supplied, which won't break the CI because we always specify the channel.

It might be easier to review the commits individually.

### CODE REVIEW

File: tools/wpt/browser.py [hunk 1]
```diff
@@ -63,6 +63,18 @@ class Browser(object):
     def __init__(self, logger):
         self.logger = logger
 
+    def _get_dest(self, dest, channel):
+        if dest is None:
+            # os.getcwd() doesn't include the venv path
+            dest = os.path.join(os.getcwd(), "_venv")
```
Comment (LukeZielinski) [inline tools/wpt/browser.py:59]: More for my own info - isn't the _venv dir suffixed with the py version these days? Curious why it doesn't care about that here.

File: tools/wpt/browser.py [hunk 2]
```diff
@@ -13,7 +12,7 @@
 from six.moves.urllib.parse import urlsplit
 import requests
 
-from .utils import call, get, untar, unzip
+from .utils import call, get, untar, unzip, rmtree
```
Comment (LukeZielinski) [inline tools/wpt/browser.py:15]: nit: keep the list alphabetical

File: tools/wpt/tests/test_wpt.py [hunk 3]
```diff
@@ -16,7 +16,7 @@
 
 import pytest
 
-from tools.wpt import wpt
+from tools.wpt import wpt, utils
```
Comment (LukeZielinski) [inline tools/wpt/tests/test_wpt.py:19]: nit: alphabetical here too

File: tools/wpt/browser.py [hunk 4]
```diff
@@ -575,7 +587,26 @@ def chromium_platform_string(self):
 
         return platform
 
+    def _chromium_package_name(self):
+        return "chrome-%s" % self.platforms.get(uname[0]).lower()
+
+    def _latest_chromium_snapshot_url(self):
+        # Make sure we use the same revision in an invocation.
+        if self._last_change is None:
+            arch = self._chromium_platform_string()
```
Comment (LukeZielinski) [inline tools/wpt/browser.py:596]: nit: what is `arch` short for? archive? architecture? Wonder if another word would be clearer.

**Review by LukeZielinski — APPROVED** [review]
lgtm%nits

File: tools/wpt/browser.py (hunk 1 above)
Comment (Hexcles) [reply tools/wpt/browser.py:59]: Yeah it is, and AFAICT, `dest` is never `None` when executed from `wpt run` or `wpt install`, but I'll leave it alone for now until I do some further investigation for a cleanup/refactoring.

File: tools/wpt/browser.py (hunk 4 above)
Comment (Hexcles) [reply tools/wpt/browser.py:596]: It's architecture. I'll change the code.

File: tools/wpt/browser.py (hunk 2 above)
Comment (Hexcles) [reply tools/wpt/browser.py:15]:
```suggestion
from .utils import call, get, rmtree, untar, unzip
```

File: tools/wpt/tests/test_wpt.py (hunk 3 above)
Comment (Hexcles) [reply tools/wpt/tests/test_wpt.py:19]:
```suggestion
from tools.wpt import utils, wpt
```

File: tools/wpt/browser.py [hunk 5]
```diff
@@ -575,7 +587,26 @@ def chromium_platform_string(self):
...
+            revision_url = "https://storage.googleapis.com/chromium-browser-snapshots/%s/LAST_CHANGE" % arch
+            self._last_change = get(revision_url).text.strip()
+        return "https://storage.googleapis.com/chromium-browser-snapshots/%s/%s/" % (arch, self._last_change)
```
Comment (Hexcles) [inline tools/wpt/browser.py:599]:
```suggestion
        architecture = self._chromium_platform_string()
        if self._last_change is None:
            revision_url = "https://storage.googleapis.com/chromium-browser-snapshots/%s/LAST_CHANGE" % architecture
            self._last_change = get(revision_url).text.strip()
        return "https://storage.googleapis.com/chromium-browser-snapshots/%s/%s/" % (architecture, self._last_change)
```

**Comment by stephenmcgruer** [comment]
Drive-by: what are these runs treated as when they reach wpt.fyi? Will they be selectable separately from 'experimental' (which is what dev is currently tagged as). Will they be mixed in with dev?

**Comment by Hexcles** [comment]
I definitely don't plan to mix it with dev for various reasons (it's not the same channel, or even the same "product" in a strict sense).

For now, these runs will be more like a trial for enabling MojoJS until the next branch point (when mojojs.zip will be available for Dev) and will be given either special labels or a different product name ("chromium" instead of "chrome") on wpt.fyi to hide it from the homepage by default. They'll be scheduled less frequently (maybe daily. We may likely later find them more useful than just experimenting with MojoJS, and then we can revisit this (maybe even replace dev with Chromium ToT on wpt.fyi) as a separate discussion.

**Comment by stephenmcgruer** [comment]
Right, I'm glad that that is the intent, but I also want to make sure we've thought through whether that is what will happen.

For example, looking at wpt.fyi APIs, these are the tags for Chrome Dev and Firefox Nightly runs from the Taskcluster master push:

```
["experimental","firefox","master","nightly","taskcluster","user:someone"]
["chrome","dev","experimental","master","taskcluster","user:someone"]
```

Now, if we assume that Chromium ToT would look like (because the results-processor maps nightly to experimental):

`["chrome","nightly","experimental","master","taskcluster","user:someone"]`

Then it's important that the wpt.fyi front-end knows how to distinguish between Chrome Dev and Nightly, because if it just uses 'experimental' we're going to mix results.

Does that make sense? Apologies if this has already been thought through.

**Comment by Hexcles** [comment]
Oh sorry I forgot to mention that a change is needed in wpt.fyi, too, which is also partly why this PR does not actually schedule anything. But thanks for double checking!

**Pull request merged**